### PR TITLE
Support more formatting codes

### LINKED ISSUE #1153: [develop] viewing some rpacks like classic 3d from cf causes a segmentation fault
### Operating System

Windows, macOS, Linux, Other

### Version of PolyMC

develop

### Version of Qt

Qt 6.3.0

### Description of bug

viewing some rpacks like classic 3d from cf causes a segmentation fault

### Steps to reproduce

- download classic-3d from cf, e.g. https://www.curseforge.com/minecraft/texture-packs/classic-3d/files/3898577
- add it to the rpack section of an instance
- click on it

### Suspected cause

weird symbols in the zip name maybe?

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Scrumplex**
Closes #1153

Maybe the code can be cleanup up a bit. Seems a bit too convoluted.

### CODE REVIEW

**Comment by DioEgizio** [comment]
this definetly fixes https://github.com/PolyMC/PolyMC/issues/1153

File: launcher/ui/widgets/InfoFrame.cpp [hunk 1]
```diff
@@ -97,48 +97,60 @@ void InfoFrame::updateWithResource(const Resource& resource)
...
-    // We have to manually set the colors for use.
-    //
-    // A color is set using §x, with x = a hex number from 0 to f.
-    //
-    // We traverse the description and, when one of those is found, we create
-    // a span element with that color set.
-    //
-    // TODO: Make the same logic for font formatting too.
-    // TODO: Wrap links inside <a> tags
+	// We have to manually set the colors for use.
```
Comment (Scrumplex) [inline launcher/ui/widgets/InfoFrame.cpp]: oops. Tabs

File: launcher/ui/widgets/InfoFrame.cpp [hunk 2]
```diff
@@ -97,48 +97,60 @@ void InfoFrame::updateWithResource(const Resource& resource)
...
+	// TODO: Wrap links inside <a> tags
+
+	// https://www.sportskeeda.com/minecraft-wiki/color-codes
+	const QMap<QChar, QString> color_codes_map = {
+		{'0', "#000000"}, {'1', "#0000AA"}, {'2', "#00AA00"}, {'3', "#00AAAA"}, {'4', "#AA0000"},
+		{'5', "#AA00AA"}, {'6', "#FFAA00"}, {'7', "#AAAAAA"}, {'8', "#555555"}, {'9', "#5555FF"},
+		{'a', "#55FF55"}, {'b', "#55FFFF"}, {'c', "#FF5555"}, {'d', "#FF55FF"}, {'e', "#FFFF55"},
+		{'f', "#FFFFFF"}
+	};
+	const QMap<QChar, QString> formatting_codes_map = {
```
Comment (flowln) [inline launcher/ui/widgets/InfoFrame.cpp]:
```suggestion
	// https://minecraft.fandom.com/wiki/Formatting_codes#Formatting_codes
	const QMap<QChar, QString> formatting_codes_map = {
```

leave a reference to future maintainers! (we :slightly_smiling_face:)

**Review by flowln — APPROVED** [review]
thumbsup

**Review by timoreo22 — APPROVED** [review]

**Pull request merged**